phase1-coordinator/src/coordinator_state.rs

```diff
@@ -939,8 +939,8 @@ pub struct CoordinatorState {
     /// The current status of the coordinator.
     status: CoordinatorStatus,
     /// The map of queue participants with a reliability score, an assigned future
-    /// round, and a last seen timestamp.
-    queue: HashMap<Participant, (u8, Option<u64>, DateTime<Utc>)>,
+    /// round, a last seen timestamp, and their time of joining.
+    queue: HashMap<Participant, (u8, Option<u64>, DateTime<Utc>, DateTime<Utc>)>,
```

Comment:
perhaps it's worth refactoring this tuple out into its own struct at this point so the fields can be named?